src/ActionList/ActionList.examples.stories.tsx

```diff
@@ -187,10 +189,19 @@ export function AsyncListWithSpinner(): JSX.Element {
         filter. This pattern can be found in branch selection menus via the SelectPanel component.
       </p>
 
-      <TextInput onChange={filter} placeholder="Search branches" sx={{m: 2, mb: 0, width: 'calc(100% - 16px)'}} />
-      {results.length === 0 ? (
-        <Text sx={{display: 'block', fontSize: 1, m: 2}}>No branches match that query</Text>
-      ) : null}
+      <FormControl sx={{m: 2, mb: 0, width: 'calc(100% - 16px)'}}>
+        <FormControl.Label>Search branches</FormControl.Label>
+        <TextInput onChange={filter} block />
+      </FormControl>
+      <LiveRegion>
+        <LiveRegionOutlet isVisible={results.length === 0} />
+        {results.length === 0 ? (
+          <Text sx={{display: 'block', fontSize: 1, m: 2}}>No branches match that query</Text>
+        ) : (
+          `${results.length} branches match that query`
+        )}
+      </LiveRegion>
```

Comment:
Ah, sorry I confused this earlier where the message here _would_ be visible (I totally misread the `VisuallyHidden` helper 🤦‍♂️)

The LiveRegion module currently is intended to mirror the helper that exists upstream in dotcom in that it's a visually hidden part of the page. Specifically, things passed as `children` to the `Message` component are analogous to the "announce" helper.

The important part about this is because, for certain live regions types, they must _always_ be in the DOM as dynamically adding them can cause certain Screen Readers to not announce as expected 😞 

For visible content that also is considered a live region, it would be worth having helpers for the status/alert roles (or just practices to follow). In this case, it could be:

```suggestion
      <div role="status">
        {results.length === 0
          ? <Text sx={{display: 'block', fontSize: 1, m: 2}}>No branches match that query</Text>
          : <VisuallyHidden>`${results.length} branches match that query`</VisuallyHidden>
      </div>
```

Would need to test to double-check, but I think that should satisfy the same thing as `aria-live="polite" aria-atomic="true"` and would keep the live region persistent in the DOM. Curious what you think @mperrotti and @ericwbailey 👀